pkg/process/metadata/parser/java/yaml.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package javaparser
+
+import (
+	"io"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+type flattener struct {
+	props map[string]string
+}
+
+func (f *flattener) visit(path []string, node *yaml.Node) {
+	for i := 0; i < len(node.Content); i += 2 { // should be safer
```

Comment:
perhaps the comment was referring to the fact that I assumed that's even(as you commented after below). I'll check that parity and skip if not matching to avoid bad accesses here